src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -105,72 +105,67 @@ public static bool SequenceEqual(ref byte first, ref byte second, int length)
             int index = 0;
```

Comment:
You would have to pin it first. `Unsafe.AsPointer` is safe to use only if you know that the `ref` is pinned.